src/convertToObject.js

```diff
@@ -6,7 +6,16 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((value) => value.trim())
+    .filter((value) => value !== '')
+    .reduce((prev, string) => {
+      return {
+        ...prev,
+        [string.split(':')[0].trim()]: string.split(':')[1].trim(),
```

Comment:
[CODE STYLE] - Avoid using array index access directly after a split operation as it can be error-prone and hard to read. Consider destructuring the result of the split into meaningful variable names to improve readability and maintainability.